linera-graphql-client/src/service.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use graphql_client::GraphQLQuery;
+use linera_base::{
+    crypto::CryptoHash,
+    data_types::{Amount, BlockHeight, Timestamp},
+    identifiers::{ChainId, Destination, Owner},
+};
+
+#[cfg(target_arch = "wasm32")]
+mod types {
+    use super::{BlockHeight, ChainId, CryptoHash};
+    use linera_base::data_types::RoundNumber;
+    use serde::{Deserialize, Serialize};
+    use serde_json::Value;
+
+    pub type Epoch = Value;
+    pub type Message = Value;
+    pub type Operation = Value;
+    pub type Event = Value;
+    pub type Origin = Value;
+    pub type UserApplicationDescription = Value;
+    pub type ApplicationId = String;
+
+    #[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+    pub struct Notification {
+        pub chain_id: ChainId,
+        pub reason: Reason,
+    }
+
+    #[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+    #[allow(clippy::enum_variant_names)]
+    pub enum Reason {
+        NewBlock {
+            height: BlockHeight,
+            hash: CryptoHash,
+        },
+        NewIncomingMessage {
+            origin: Origin,
+            height: BlockHeight,
+        },
+        NewRound {
+            height: BlockHeight,
+            round: RoundNumber,
+        },
+    }
```

Comment:
This all looks like a good reason to move some more types to `linera_base` or make `linera_chain` wasm32-compatible… not sure. Maybe there's already some ticket to track something like this?

Reply:
I agree, but it was too much change to put in this commit

Reply:
Makes sense. But it would be good to create an issue, and link to it from here.